pkg/scheduler/framework/frameworkutils.go

```diff
@@ -187,19 +188,27 @@ func crossReferencePickedCustersAndObsoleteBindings(
 
 // newSchedulingDecisionsFrom returns a list of scheduling decisions, based on the newly manipulated list of
 // bindings and (if applicable) a list of filtered clusters.
-func newSchedulingDecisionsFrom(maxClusterDecisionCount int, filtered []*filteredClusterWithStatus, existing ...[]*fleetv1beta1.ClusterResourceBinding) []fleetv1beta1.ClusterDecision {
+func newSchedulingDecisionsFrom(maxUnselectedClusterDecisionCount int, filtered []*filteredClusterWithStatus, existing ...[]*fleetv1beta1.ClusterResourceBinding) []fleetv1beta1.ClusterDecision {
 	// Pre-allocate with a reasonable capacity.
-	newDecisions := make([]fleetv1beta1.ClusterDecision, 0, maxClusterDecisionCount)
+	newDecisions := make([]fleetv1beta1.ClusterDecision, 0, maxUnselectedClusterDecisionCount)
 
 	// Build new scheduling decisions.
+	slotsLeft := clustersDecisionArrayLengthLimitInAPI
 	for _, bindingSet := range existing {
-		for _, binding := range bindingSet {
-			newDecisions = append(newDecisions, binding.Spec.ClusterDecision)
+		setLength := len(bindingSet)
+		for i := 0; i < setLength && i < slotsLeft; i++ {
+			newDecisions = append(newDecisions, bindingSet[i].Spec.ClusterDecision)
+		}
+
+		slotsLeft -= setLength
+		if slotsLeft <= 0 {
+			klog.V(2).InfoS("Reached API limit of cluster decision count; decisions off the limit will be discarded")
+			break
 		}
 	}
 
 	// Move some decisions from unbound clusters, if there are still enough room.
-	if diff := maxClusterDecisionCount - len(newDecisions); diff > 0 {
+	if diff := maxUnselectedClusterDecisionCount - len(newDecisions); diff > 0 {
```

Comment:
now the maxUnselectedClusterDecisionCount is renamed. should we update the logic here?

based on the comment, 

"// maxUnselectedClusterDecisionCount controls the maximum number of decisions for unselected clusters
	// added to the policy snapshot status."

 If i understand it correctly, we need to populate min(maxUnselectedClusterDecisionCount, slotsLeft) using filteredClusters?

Reply:
Not really. As the comment has explained, the limit is added to make sure that we only populate the status with info. for filtered out clusters if the total number of decisions is small enough; i.e. we won't fill all the slots blindly with decisions for unfiltered clusters.

Reply:
i feel the original field name is better. I was confused by the new name and its meaning. :)